constants.py

```diff
@@ -63,6 +71,25 @@
 """
 
 
+SPECIAL_TOKENS_FOR_TRAIL = {i : NUMBER_OF_EACH_SPECIAL_FOOD_TOKEN for i in SPECIAL_FOOD_TYPES}
+"""The standard number of each special food token
+
+A dictionary showing how many of each special food token will be 
+used as per the rules.
+Keys are foods as strings
+Values are integers
+"""
+
+
+TOKENS_FOR_TRAIL = dict(STANDARD_TOKENS_FOR_TRAIL, **SPECIAL_TOKENS_FOR_TRAIL)
```

Comment:
You might be interested to know that from Python 3.9 (released just a few days ago on the 17th Oct 2020), the combining of two dictionaries has been simplified. See https://stackoverflow.com/questions/38987/how-do-i-merge-two-dictionaries-in-a-single-expression-in-python-taking-union-o

Reply:
Yes, I actually read some of this page earlier.

Reply:
Note how the stack overflow link suggests `{**STANDARD_TOKENS_FOR_TRAIL, **SPECIAL_TOKENS_FOR_TRAIL}` with `**` in the first one too. That seems to be required when using `{}` but not when using `dict()` (although it does work too if you use the `**` in the first argument of the `dict()`).